cpp/include/kvikio/defaults.hpp

```diff
@@ -27,7 +27,47 @@
 #include <kvikio/shim/cufile.hpp>
 
 namespace kvikio {
+/**
+ * @brief I/O compatibility mode.
+ */
+enum class CompatMode : uint8_t {
+  OFF,   // Enforce cuFile I/O. GDS will be activated if the system requirements for cuFile are met
+         // and cuFile is properly configured. However, if the system is not suited for cuFile, I/O
+         // operations under the OFF option may error out, crash or hang.
+  ON,    // Enforce POSIX I/O.
+  AUTO,  // Try cuFile I/O first, and fall back to POSIX I/O if the system requirements for cuFile
+         // are not met.
+};
+
 namespace detail {
+/**
+ * @brief Parse a string into a CompatMode enum.
+ *
+ * @param compat_mode_str Compatibility mode in string format. Valid values include "ON", "OFF",
+ * "AUTO" (case-insensitive).
+ * @return A CompatMode enum.
+ */
+inline CompatMode parse_compat_mode_str(std::string_view compat_mode_str)
+{
+  // Convert to lowercase
+  std::string compat_mode_lower{compat_mode_str};
+  std::transform(compat_mode_lower.begin(),
+                 compat_mode_lower.end(),
+                 compat_mode_lower.begin(),
+                 [](unsigned char c) { return std::tolower(c); });
+
+  CompatMode res{};
+  if (compat_mode_lower == "on") {
+    res = CompatMode::ON;
+  } else if (compat_mode_lower == "off") {
```

Comment:
Should we still support `1/0, true/false, yes/no` ?

Reply:
Completed. The unit test is updated.